File: 02/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdbool.h>

int countLetters(char text[]);
int countWords(char text[]);
int countSentences(char text[]);


int main(void) {

    char text[1000];

    printf("Enter Text: ");
    fgets(text, sizeof(text), stdin);

    int letters = countLetters(text);
    int words = countWords(text);
    int sentences = countSentences(text);

    int gradeLevel = roundf(0.0588 * (((float) letters / (float) words) * 100.0) - 0.296 * (((float) sentences / (float) words) * 100.0) - 15.8);

    if(index >= 16) {
        printf("Grade 16+\n");
    }
    else if(index < 1) {
        printf("Before Grade 1\n");
    }
    else {
        printf("Grade %i\n", index);
    }

}

int countLetters(char text[]) {

    int textLength = strlen(text);
    int count = 0;

    for(int textIndex = 0; textIndex < textLength; textIndex++) {
        bool characterIsAlphabetical = (text[textIndex] >= 'A' && text[textIndex] <= 'Z') || (text[textIndex] >= 'a'&& text[textIndex] <= 'z');

        if(characterIsAlphabetical) {
            count++;
        }
    }

    return count;
}

int countWords(char text[]) {

    int textLength = strlen(text);
    int count = 0;

    for(int textIndex = 0; textIndex < textLength; textIndex++) {
        
        bool characterIsSpace = text[textIndex] == ' ';

        if(characterIsSpace) count++;
    }

    if(count == 0) count++;

    return count += 1;
}

int countSentences(char text[]) {

    int textLength = strlen(text);
    int count = 0;

    for(int textIndex = 0; textIndex< textLength; textIndex++) {

        bool characterIsExclamationMark = text[textIndex] == '!';
        bool characterIsQuestionMark = text[textIndex] == '?';
        bool characterIsPeriod = text[textIndex] == '.';

        if(characterIsExclamationMark || characterIsQuestionMark || characterIsPeriod) {
            count++;
        }
    }

    return count;
}

File: 06/sentimental-readability/readability.py
def main():
    text = input("Enter Text: ")

    letters = countLetters(text)
    words = countWords(text)
    sentences = countSentences(text)

    gradeLevel = round(0.0588 * (( letters / words) * 100.0) - 0.296 * ((sentences / words) * 100.0) - 15.8)

    if gradeLevel >= 16:
        print("Grade 16+")
    elif gradeLevel < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {gradeLevel}")

def countLetters(text):
    textLength = len(text)
    count = 0

    for textIndex in range(textLength):

        if text[textIndex].isalpha():
            count += 1

    return count

def countWords(text):
    textLength = len(text)
    count = 0

    for textIndex in range(textLength):
        characterIsSpace = text[textIndex] == ' '

        if characterIsSpace:
            count += 1
    
    return count + 1

def countSentences(text):
    textLength = len(text)
    count = 0

    for textIndex in range(textLength):
        characterIsExclamationMark = text[textIndex] == '!'
        characterIsQuestionMark = text[textIndex] == '?'
        characterIsPeriod = text[textIndex] == '.'

        if characterIsExclamationMark or characterIsQuestionMark or characterIsPeriod:
            count += 1

    return count

main()